gossip/peer_test.go

```diff
@@ -0,0 +1,67 @@
+package gossip
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestPeerProgressWithEpoch(t *testing.T) {
+	// Increment epoch and see if the peer is progressing
+	setProgressThreshold(1 * time.Second)
+	newPeer := getPeer()
+	ep1 := PeerProgress{Epoch: 1}
+	newPeer.SetProgress(ep1)
+	time.Sleep(2 * time.Second) //set the threshold to 2 second
```

Comment:
All these `Sleep` acctumulate to 9 seconds - making test runs 9 seconds slower as I understand. Isn't there a different way to test this? Do we actually even need to sleep?